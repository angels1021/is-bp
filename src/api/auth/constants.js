/**
 * auth saga constants.
 */

export const LOGIN_REQUEST = 'api/auth/LOGIN_REQUEST';
export const LOGOUT_REQUEST = 'api/auth/LOGOUT_REQUEST';
export const AUTH_REQUEST = 'api/auth/AUTH_REQUEST';
export const AUTH_ERROR = 'api/auth/AUTH_ERROR';
export const AUTH_SUCCESS = 'api/auth/AUTH_SUCCESS';
export const AUTH_SET = 'api/auth/AUTH_SET';

export const initialState = {
  username: '',
  password: '',
  location: '/',
  code: ''
};
