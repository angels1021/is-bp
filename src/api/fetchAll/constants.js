/**
 * fetchAll api constants
 */

export const FETCH_ALL = 'api/fetchAll/FETCH_ALL';
