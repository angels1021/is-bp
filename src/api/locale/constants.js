/**
 * Locale api shared constants.
 */

export const CHANGE_LOCALE_REQUEST = 'api/locale/CHANGE_LOCALE_REQUEST';
export const CHANGE_LOCALE_SUCCESS = 'api/locale/CHANGE_LOCALE_SUCCESS';
export const CHANGE_LOCALE_FAIL = 'api/locale/CHANGE_LOCALE_FAIL';
export const CHANGE_LOCALE_CANCEL = 'api/locale/CHANGE_LOCALE_CANCEL';

