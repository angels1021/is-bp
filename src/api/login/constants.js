/*
 *  Login/Logout Sagas constants
 *
 * */

export const CHANGE_FORM = 'api/login/CHANGE_FORM';
export const SET_AUTH = 'api/login/SET_AUTH';
export const REQUEST_PENDING = 'api/login/REQUEST_PENDING';
export const LOGIN_REQUEST = 'api/login/LOGIN_REQUEST';
export const LOGOUT = 'api/login/LOGOUT';
export const REQUEST_SUCCESS = 'api/login/REQUEST_SUCCESS';
export const REQUEST_ERROR = 'api/login/REQUEST_ERROR';
export const CLEAR_ERROR = 'api/login/CLEAR_ERROR';

export const initialState = {
  username: '',
  password: '',
  location: '/',
  code: ''
};
