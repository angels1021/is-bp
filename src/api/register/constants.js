/*
 *  Register Sagas constants
 *
 * */

export const CHANGE_FORM = 'api/register/CHANGE_FORM';
export const SET_AUTH = 'api/register/SET_AUTH';
export const SENDING_REQUEST = 'api/register/SENDING_REQUEST';
export const REGISTER_REQUEST = 'api/register/REGISTER_REQUEST';
export const REQUEST_ERROR = 'api/register/REQUEST_ERROR';
export const CLEAR_ERROR = 'api/register/CLEAR_ERROR';

export const cleanForm = {
  username: '',
  password: ''
};
