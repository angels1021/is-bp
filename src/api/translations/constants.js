/**
 * Constants for the the translations api
 */

export const MESSAGES_REQUEST = 'api/translations/MESSAGES_REQUEST';
export const MESSAGES_SUCCESS = 'api/translations/MESSAGES_SUCCESS';
export const MESSAGES_FAIL = 'api/translations/MESSAGES_FAIL';
