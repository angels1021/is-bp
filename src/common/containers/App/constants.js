/*
* app state constants
*/

export const DEFAULT_LOCALE = 'en';

export const FETCH_ALL = (moduleId) => `routes/${moduleId}/FETCH_ALL`;

export const ASYNC_REQUEST = 'app/common/ASYNC_REQUEST';
export const ASYNC_SUCCESS = 'app/common/ASYNC_SUCCESS';
export const ASYNC_FAIL = 'app/common/ASYNC_FAIL';
