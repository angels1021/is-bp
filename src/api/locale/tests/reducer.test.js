/**
 * Test locale api reducer
 */

import LocaleReducerFactory from '../reducer';
import {
  changeLocale,
  changeLocaleFail,
  changeLocaleSuccess,
  changeLocaleCancel
} from '../actions';
import { CHANGE_LOCALE_REQUEST } from '../constants';

const moduleId = 'apiTest';
const localeReducer = LocaleReducerFactory(moduleId);

describe('API locale reducer', () => {
  let state;
  const request = 'he';
  beforeEach(() => {
    state = localeReducer(undefined, {});
  });

  it('should return the initial state', () => {
    expect(state).toMatchSnapshot();
  });

  it('should return the initial state if moduleId doesn\'t match', () => {
    const result = localeReducer(state, {
      type: CHANGE_LOCALE_REQUEST,
      payload: { locale: request, moduleId: 'someOtherModule' }
    });

    expect(result).toMatchSnapshot();
  });

  describe('CHANGE_LOCALE_REQUEST', () => {
    it('should reset error and set pending to requested locale', () => {
      // assert
      const expectedResult = state
        .set('error', false)
        .set('pending', request);
      // act
      const result = localeReducer(state, changeLocale(request, [], moduleId));
      // assert
      expect(result).toEqual(expectedResult);
    });
  });

  describe('CHANGE_LOCALE_SUCCESS', () => {
    it('should reset error  and pending and set locale to request', () => {
      // assert
      const expectedResult = state
        .set('error', false)
        .set('locale', request)
        .set('pending', false);
      // act
      const result = localeReducer(state, changeLocaleSuccess(request, moduleId));
      // assert
      expect(result).toEqual(expectedResult);
    });
  });

  describe('CHANGE_LOCALE_FAIL', () => {
    it('should reset pending and set error to response error', () => {
      // assert
      const error = new Error('some error');
      const expectedResult = state
        .set('error', error)
        .set('pending', false);
      // act
      const result = localeReducer(state, changeLocaleFail(error, moduleId));
      // assert
      expect(result).toEqual(expectedResult);
    });
  });

  describe('CHANGE_LOCALE_CANCEL', () => {
    it('should reset pending and error', () => {
      // assert
      const expectedResult = state
        .set('error', false)
        .set('pending', false);
      // act
      const result = localeReducer(state, changeLocaleCancel(moduleId));
      // assert
      expect(result).toEqual(expectedResult);
    });
  });
});
