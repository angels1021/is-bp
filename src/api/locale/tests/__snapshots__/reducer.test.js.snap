// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`API locale reducer should return the initial state 1`] = `
Object {
  "error": false,
  "locale": "en",
  "pending": false,
}
`;

exports[`API locale reducer should return the initial state if moduleId doesn't match 1`] = `
Object {
  "error": false,
  "locale": "en",
  "pending": false,
}
`;
